docs/source/services/timelock_service/reverse-migration.rst

```diff
@@ -0,0 +1,140 @@
+.. _timelock-reverse-migration:
+
+Reverse Migration
+=================
+
+.. danger::
+
+   AtlasDB does not natively support reverse migrations. Improperly executing reverse migration from external timestamp
+   and lock services can result in **SEVERE DATA CORRUPTION**! Please contact the AtlasDB team before attempting a
+   reverse migration.
+
+If one wishes to downgrade from an external TimeLock Server to embedded timestamp and lock services, one can perform
+the inverse of the database migrations mentioned in :ref:`manual-timelock-migration`. It is also important to update the
+embedded timestamp bound to account for any timestamps issued since the original migration.
+
+The reverse migration process must be run offline (with no AtlasDB clients running) and consists of the following steps.
+Throughout this document, we assume the AtlasDB client you are trying to reverse-migrate is ``client`` .
+
+#. Shut down your AtlasDB clients.
+#. Take a restore timestamp.
+#. Update your key-value service to the restore timestamp.
+#. Revert your AtlasDB client configuration.
+#. Shut down the TimeLock servers.
+#. Remove your client from the list of TimeLock clients.
+#. Restart your AtlasDB clients.
+#. Restart the TimeLock servers.
```

Comment:
Yep. I've changed the final lineup so that it's closer to this (though not precisely since there is a bit of machinery needed to do the client config right). The final solution has the same amount of downtime I think in that the client is down for the entire upgrade but timelock is down for a single, immediate-ish timelock bounce. Thanks for the suggestion!